project-koku/koku
# 1795 openshift on all permissions

#1795 

The general idea here:
1. Access to the ocp-on-all view is granted as long as a user has access to a single provider type. This access is determined using the new `openshift_all_access`.
2. When setting access for the ocp-on-all reporting view, the data is filtered on all accounts the user has access to. If, for example, a user has access to all AWS accounts and a subset of the Azure accounts, we look up all AWS account numbers in the DB and append to the azure account number filter list. 

Testing:
1. `make create-test-customer`
2. `make load-test-customer-data start=2020-01-01 end=2020-02-24`
3. Check these endpoints, see data:
    - reports/openshift/infrastructures/all/costs/
    - reports/openshift/infrastructures/aws/costs/
    - reports/openshift/infrastructures/azure/costs/
4. Apply the following diff (This user has access to AWS/OCP but not Azure):
```
diff --git a/koku/koku/middleware.py b/koku/koku/middleware.py
index 38656124..4f4653cb 100644
--- a/koku/koku/middleware.py
+++ b/koku/koku/middleware.py
@@ -216,8 +216,8 @@ class IdentityHeaderMiddleware(MiddlewareMixin):  # pylint: disable=R0903
     def _get_access(self, user):
         """Obtain access for given user from RBAC service."""
         access = None
-        if user.admin:
-            return access
+        # if user.admin:
+        #     return access
         access = self.rbac.get_access_for_user(user)
         return access
 
@@ -284,13 +284,14 @@ class IdentityHeaderMiddleware(MiddlewareMixin):  # pylint: disable=R0903
             user.req_id = req_id
 
             cache = caches["rbac"]
-            user_access = cache.get(user.uuid)
+            # user_access = cache.get(user.uuid)
+            user_access = None
             if not user_access:
                 try:
                     user_access = self._get_access(user)
                 except RbacConnectionError as err:
                     return HttpResponseFailedDependency({"source": "Rbac", "exception": err})
-                cache.set(user.uuid, user_access, self.rbac.cache_ttl)
+                # cache.set(user.uuid, user_access, self.rbac.cache_ttl)
             user.access = user_access
             request.user = user
 
diff --git a/koku/koku/rbac.py b/koku/koku/rbac.py
index 6a5480b4..d4339d7f 100644
--- a/koku/koku/rbac.py
+++ b/koku/koku/rbac.py
@@ -225,12 +225,15 @@ class RbacService:  # pylint: disable=too-few-public-methods
             self.protocol, self.host, self.port, self.path
         )
         headers = {"x-rh-identity": user.identity_header.get("encoded")}
-        acls = self._request_user_access(url, headers)
-        if isinstance(acls, list) and len(acls) == 0:  # pylint: disable=len-as-condition
-            return None
+        # acls = self._request_user_access(url, headers)
+        # if isinstance(acls, list) and len(acls) == 0:  # pylint: disable=len-as-condition
+        #     return None
 
-        processed_acls = _process_acls(acls)
-        return _apply_access(processed_acls)
+        # processed_acls = _process_acls(acls)
+        # return _apply_access(processed_acls)
+
+        processed_acls = {'aws.account': {'read': ['*']}, 'azure.subscription_guid': {'read': []}, 'openshift.cluster': {'read': ['*']}, 'openshift.node': {'read': ['*']}, 'openshift.project': {'read': ['*']}, 'provider': {'read': ['*'], 'write': ['*']}, 'rate': {'read': [], 'write': []}}
+        return processed_acls
 
     def get_cache_ttl(self):
```
5. Check reports/openshift/infrastructures/azure/costs endpoint and see 403 permission error
6. Check aws/cost and all/cost, and see matching values.


[ocp-on-all.txt](https://github.com/project-koku/koku/files/4247051/ocp-on-all.txt)
